src/differentiation/vecjac_products.jl

```diff
@@ -14,9 +14,9 @@ function num_vecjac!(du, f, x, v, cache1 = similar(v), cache2 = similar(v);
     return du
 end
 
-function num_vecjac(f, x, v, f0 = nothing)
-    vv = reshape(v, axes(x))
-    f0 === nothing ? _f0 = f(x) : _f0 = f0
+function num_vecjac(f::F, x, v, f0 = nothing) where {F}
+    f0 === nothing ? (_f0 = f(x)) : (_f0 = f0)
```

Comment:
```suggestion
    _f0 = ifelse(isnothing(f0), f(x), f0)
```

Reply:
This will cause an unnecessary computation of `f(x)` which is not needed